### ISSUE TITLE
Move flake-compat to the NixOS org

### ISSUE THREAD

**Issue by infinisil** [nix-open-org/org-v2#49]
[flake-compat](https://github.com/edolstra/flake-compat) is currently a repo under @edolstra's account, even though it's become a core flake utility.

@edolstra Would you consider transferring flake-compat to the NixOS GitHub org, to both solidify its official status, and to let others perform or delegate its maintenance?

Note that because this would newly be an official project, we'd also [need the Steering Committee to approve it](https://github.com/NixOS/org/blob/bc1f447e204324829e757e6be05c57000665368b/doc/constitution.md?plain=1#L44).

**Comment by nyabinary** [comment]
Any progress on this?

**Comment by MattSturgeon** [comment]
[edolstra/flake-compat](https://github.com/edolstra/flake-compat) seems to go through periods of being maintained and periods of being semi-abandoned. I think this is normal for any project that is maintained by a single person.

Personally, I have two PRs open that haven't been looked at in months. Others have had PRs open for years: [flake-compat PRs](https://github.com/edolstra/flake-compat/pulls).

...Not that I wish to be ungrateful: @edolstra has no obligation to look at PRs that don't interest him. On the contrary, I'm grateful for everything he _has_ done!

#### Maintainers

Maybe the nix team itself would be willing to maintain it, given it is essentially a copy of the [`call-flake.nix`](https://github.com/NixOS/nix/blob/2.31.1/src/libflake/call-flake.nix) file included in `nix`. I was grateful to @roberth for helping [steer](https://github.com/edolstra/flake-compat/pull/71#pullrequestreview-2832125437) one of my PRs in the right direction, although it's still awaiting ACK from someone who can merge 😉.

I'd also be willing to help with some of the maintenance burden, at least where I'm not too far out of my depth technically.

#### Forks

For the record, there are forks of flake-compat, including [nix-community/flake-compat](https://github.com/nix-community/flake-compat) (whose maintenance status is also [unclear](https://github.com/nix-community/flake-compat/issues/6), but historically maintained by @zimbatm) and [lix-project/flake-compat](https://git.lix.systems/lix-project/flake-compat) (which I believe is maintained by @lf-).

It'd be great to reconcile all of these (or at least, all that are willing) as one project under the [NixOS org](https://github.com/NixOS).

#### Steering Committee

> Note that because this would newly be an official project, we'd also [need the Steering Committee to approve it](https://github.com/NixOS/org/blob/bc1f447e204324829e757e6be05c57000665368b/doc/constitution.md?plain=1#L44).

Has any progress been made discussing this with the SC?

**Comment by lf-** [comment]
The one of these in the best maintenance state afaik is the lix one :)

However it is a fork that intentionally doesn't include some changes like removing emulation of fetchTree due to nix implementation bugs. It also has tests and new features. 

The real blocker to this all being merged together is *entirely* social: the edolstra repo is used to promote flakehub and only cares about CppNix compatibility, and to try to merge these requires all parties giving up a degree of control and listening to each other, which has historically not gone well. Not to say it cannot go well, but it requires making sure both sides actually *agree* with the purpose and direction of the project in each change as well as the compat implications.

This sounds like it would make maintaining Lix flake-compat many times harder for dubious benefit to our production use of it, as it's a small nix library that does one thing and doesn't have to be uniform across the ecosystem.

**Comment by zimbatm** [comment]
Personally I'd be up to merge efforts to reduce confusion for users. The ecosystem has way too many variants of everything.

What would you need @lf- to make you feel comfortable have your version in the NixOS org? Once this becomes the official space, Eelco can still decide what he wants to do with his version.

**Comment by roberth** [comment]
> [@edolstra](https://github.com/edolstra) agreed to move https://github.com/edolstra/flake-compat to the NixOS github org so that we can extend the maintainer base.

**Comment by infinisil** [comment]
@edolstra Awesome! Then the next step is to make an org owner (e.g. me) admin of flake-compat, so the transfer can be done.

**Comment by winterqt** [comment]
Doesn’t this need SC approval first…?

**Comment by Mic92** [comment]
> Doesn’t this need SC approval first…?

Usually two org owner approvals for github related admin tasks.

**Comment by winterqt** [comment]
Nope, we’re having this discussion in an adjacent issue: https://github.com/NixOS/org/issues/168#issuecomment-3312099484

**Comment by roberth** [comment]
SC already has voted in favor, some time ago, async. Look for "Make flake-compat official"

> Make flake-compat an official NixOS org project

**Comment by winterqt** [comment]
I was going off the recent comment from @MattSturgeon that asked if we approved it, and I assumed nobody replying meant "no." I guess Silvan has a better memory than me. Sorry!

**Comment by infinisil** [comment]
Haha no I don't think I have access to the records @roberth mentioned. You're right that I should've double checked that this has SC approval and forgot :)

**Comment by lf-** [comment]
> What would you need [@lf-](https://github.com/lf-) to make you feel comfortable have your version in the NixOS org? Once this becomes the official space, Eelco can still decide what he wants to do with his version.

I'm not interested in having to be subjected to NixOS governance, so I am not interested in moving it at this time. If the project meaningfully reduced its rate of trust violating events (and friends being burned out by them) in a given year, I might consider it, however that has not been my experience.

**Comment by Ericson2314** [comment]
I approve moving it (I am a bit confused by thread here on voting, so want to make abundantly clear that I am another SC member for it).

**Comment by infinisil** [comment]
This is now done, thanks @edolstra and @Mic92!

I gave the @NixOS/nix-team admin access to the repo, that makes the most sense to me

**Comment by zimbatm** [comment]
@lf- okay. It would be good to get a clearer picture of what that looks like, but seems out of scope for this issue. I'd be happy to chat about this sometime.

I've archived
https://github.com/nix-community/flake-compat and pointed it to the official one.
